src/azure-cli/azure/cli/command_modules/network/custom.py

```diff
@@ -4782,65 +4782,102 @@ def set_lb_outbound_rule(instance, cmd, parent, item_name, protocol=None, outbou
 
 def create_lb_probe(cmd, resource_group_name, load_balancer_name, item_name, protocol, port,
                     path=None, interval=None, threshold=None, probe_threshold=None):
+    if probe_threshold is not None:
+        logger.warning(
+            "Please note that the parameter --probe-threshold is currently in preview and is not recommended "
+            "for production workloads. For most scenarios, we recommend maintaining the default value of 1 "
+            "by not specifying the value of the property."
+        )
+
     from .aaz.latest.network.lb import Show
     lb = Show(cli_ctx=cmd.cli_ctx)(command_args={
+        "name": load_balancer_name,
         "resource_group": resource_group_name,
-        "name": load_balancer_name
     })
 
-    for probe in lb["probes"]:
-        probe["interval_in_seconds"] = probe.pop("intervalInSeconds")
-        probe["number_of_probes"] = probe.pop("numberOfProbes")
-        probe["probe_threshold"] = probe.pop("probeThreshold")
-        if "requestPath" in probe:
-            probe["request_path"] = probe.pop("requestPath")
-
-    probe_args = {
+    probes = []
+    for item in lb["probes"]:
+        probe = dict()
+        probe["name"] = item.pop("name", None)
+        probe["interval_in_seconds"] = item.pop("intervalInSeconds", None)
+        probe["number_of_probes"] = item.pop("numberOfProbes", None)
+        probe["port"] = item.pop("port", None)
+        probe["probe_threshold"] = item.pop("probeThreshold", None)
+        probe["protocol"] = item.pop("protocol", None)
+        probe["request_path"] = item.pop("requestPath", None)
+        probes.append(probe)
```

Comment:
If new probe name exists in lb['probles'], do we need to raise error?

Reply:
What's the behavior before?

Reply:
Previously, we have a warning handled by `upsert_to_collection`:
https://github.com/Azure/azure-cli/blob/df34f7033b514f1f52244aaae70189821d5aedc0/src/azure-cli-core/azure/cli/core/commands/__init__.py#L486-L491